src/js/actions/layers.js

```diff
@@ -487,12 +488,14 @@ define(function (require, exports) {
                     this.dispatch(events.document.history.nonOptimistic.RESET_BOUNDS, payload);
                 }
             }).then(function () {
+                return this.transfer(guides.queryCurrentGuides);
```

Comment:
Yes, `resetBorderPolicies` transfers to `resetGuidePolicies` so we need the latest guides.